lexos/static/js/scripts_cut.js

```diff
@@ -1,382 +1,663 @@
+            Segments:</span><input type="number" min="1" step="1" \
+            name="cutValue_' + fileID + '" class="cut-text-input" \
+            id="individualCutValue" value=""/></label></div></div><div \
+            class="row overlap-div"><div class="col-md-6 pull-right" \
+            style="padding-left:2px;padding-right:3%;"><label>Overlap: \
+            <input type="number" min="0" name="cutOverlap_' + fileID + '" \
+            class="cut-text-input overlap-input" id="individualOverlap" \
+            value="0"/></label></div></div>\
+            <div id="lastprop-div_' + fileID + '" \
+            class="row lastprop-div"><div class="col-md-6 pull-right" \
+            style="padding-left:2px;padding-right:1%;"><label>Last Proportion \
+            Threshold: <input type="number" min="0" \
+            id="cutLastProp_' + fileID + '" \
+            name="cutLastProp_' + fileID + '" \
+            class="cut-text-input lastprop-input" value="50" \
+            style="width:54px;margin-right:3px;"/> %</label></div></div><div \
+            class="row"><div class="col-md-6 pull-right" \
+            style="padding-left:2px;padding-right:1%;">\
+            <label>Cutset Label: <input type="text" \
+            name="cutsetnaming_' + fileID + '" class="cutsetnaming" \
+            value="' + filename + '" style="width:155px;display:inline; \
+            margin: auto; white-space: nowrap; overflow: hidden; \
+            text-overflow: ellipsis;></label></div></div><div \
+            class="row cuttingdiv" id="cutByMSdiv"><div \
+            class="col-sm-4"><label><input type="checkbox" \
+            class="indivMS" name="cutByMS_' + fileID + '" \
+            id="cutByMS_' + fileID + '"/>Cut by Milestone</label></div><div \
+            class="col-sm-8 pull-right" id="MSoptspan" \
+            style="display:none;"><span>Cut document on this term \
+            <input type="text" class="indivMSinput" \
+            name="MScutWord_' + fileID + '" id="MScutWord' + fileID + '" \
+            value="" \
+            style="margin-left:3px;width:130px;"/></span></div>\
+            </div></fieldset></div>'
+}
+
+// process
+/**
+ * checks the form data for errors and warnings
+ * @param {string} action
+ */
 function process (action) {
-  $('#status-prepare').css({ 'visibility': 'visible', 'z-index': '400000' })
-  $('#formAction').val(action)
-  $.when(checkForErrors()).done(function () {
-    if ($('#hasErrors').val() == 'false') {
-      checkForWarnings()
-      $.when(checkForWarnings()).done(function () {
-        if ($('#needsWarning').val() == 'false') {
-          doAjax(action)
+    $('#status-prepare').css({'visibility': 'visible', 'z-index': '400000'})
+    $('#formAction').val(action)
+    $.when(checkForErrors()).done(function () {
+        if ($('#hasErrors').val() === 'false') {
+            checkForWarnings()
+            $.when(checkForWarnings()).done(function () {
```

Comment:
can't you just return the warnings from the `checkForWarning` function? so that you don't need to get all those magical hidden divs floating around your webpage.